contracts/Launchpeg.sol

```diff
@@ -351,15 +382,18 @@ contract Launchpeg is BaseLaunchpeg, ILaunchpeg {
         atPhase(Phase.PublicSale)
     {
         if (
-            totalSupply() + _quantity >
+            _numberMintedWithPreMint(msg.sender) + _quantity >
+            maxPerAddressDuringMint
+        ) {
+            revert Launchpeg__CanNotMintThisMany();
+        }
+        if (
+            _totalSupplyWithPreMint() + _quantity >
```

Comment:
@Max-3-7 @Mathieu-Be i noticed the public sale checks in `Launchpeg` and `FlatLaunchpeg` differ slightly. in `Launchpeg` we check that the public sale amount cannot take up the dev allocation (if project dev has not fully minted the allocation by the public sale phase), but not in `FlatLaunchpeg`. just wondering if this is intended (or perhaps i misunderstood the code)?